packages/shared/src/hooks/streaks/useStreakRecover.ts

```diff
@@ -83,7 +85,9 @@ export const useStreakRecover = ({
     queryFn: async () => {
       const res = await gqlClient.request(USER_STREAK_RECOVER_QUERY);
 
-      if (!res?.streakRecover?.canRecover && !!streakRestore) {
+      const userCantRecoverInNotificationCenter =
+        !res?.streakRecover?.canRecover && !!streakRestore;
```

Comment:
Ah sorry I meant you could actually pass it to the hook as `enabled`?
then it can be whatever from top level already so you dont need secondary things inside here.

Reply:
res comes from API call so we need here. We can pass from outside `streakRestore` but the API call need to be fired even if it is _true_, we need the check just to open the toast only in the notification center @rebelchris

Reply:
But you already have it outside where you iniate the hook? 
Eg listening for this specific router param or from boot where you have `canRecover` 

Not sure i'm being clear with what I'm seeing?

Reply:
Yes, but this handle the Viktor's edge case, where you have boot alert set to true (and modal opening in background) and you keep seeing the toast. In this case you want user to not see the toast!

Reply:
PS: I've put here because if I use displayToast inside a useEffect it makes the component rerender in a infinite loop :(